app/Knights/knight.py

```diff
@@ -0,0 +1,32 @@
+from app.Staff.armour import Armour
+from app.Staff.potion import Potion
+from app.Staff.weapon import Weapon
+
+
+class Knight:
+    knights = {}
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            health_points: int,
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.health_points = health_points
+        self.protection = 0
+        Knight.knights = {self.name: self}
+
+    def knight_stats_completing(
+            self,
+            armours: list[Armour],
+            weapon: Weapon,
+            potion: Potion
+    ) -> None:
+        self.power += weapon.power + potion.power
+        self.protection += sum([
+            armour.protection for armour in armours
+            if armours != []

```

Comment:
The condition `if armours != []` is unnecessary here. The list comprehension will naturally handle an empty list without needing this check. You can remove this condition.